src/main/scala/everywhere/e300artydevkit/E31-Core.scala

```diff
@@ -0,0 +1,51 @@
+package sifive.freedom.everywhere.e300artydevkit
+
+import Chisel._
+import freechips.rocketchip.config._
+import freechips.rocketchip.devices.debug._
+import freechips.rocketchip.devices.tilelink._
+import freechips.rocketchip.diplomacy._
+import freechips.rocketchip.rocket._
+import freechips.rocketchip.tile._
+import freechips.rocketchip.tilelink._
+import freechips.rocketchip.util._
+import freechips.rocketchip.subsystem._
+import freechips.rocketchip.system._
+
+
+class E31Config extends Config(
+  new WithNoMemPort ++
+  new WithNMemoryChannels(0) ++
+  new WithNBanks(0) ++
+  new With1E31Core ++
+  new BaseConfig
+)
+
+class With1E31Core extends Config((site, here, up) => {
+  case XLen => 32
+  case RocketTilesKey => List(RocketTileParams(
```

Comment:
I don't like this method of completely copying the RocketTilesKey and only changing a few parameters, I think a cleaner way to do it is to use the `alter` and `copy` functions like [here](https://github.com/sifive/freedom/blob/3624efff1819e52cec30c72f9085158189f8b53f/src/main/scala/unleashed/DevKitConfigs.scala#L41) to copy the whole preexisting key and just change the necessary parameters.

Thanks!